src/pudl/output/pudltabl.py

```diff
@@ -339,173 +386,11 @@ def gen_eia923(self, update: bool = False) -> pd.DataFrame:
-        min_cap_fact: float = 0.0,
-        max_cap_fact: float = 1.5,
-        all_gens: bool = True,
-        gens_cols: Literal["all"] | list[str] | None = None,
-    ) -> pd.DataFrame:
-        """Calculate and return generator level MCOE based on EIA data.
-
-        Eventually this calculation will include non-fuel operating expenses
-        as reported in FERC Form 1, but for now only the fuel costs reported
-        to EIA are included. They are attibuted based on the unit-level heat
-        rates and fuel costs.
-
-        Args:
-            update: If True, re-calculate dataframe even if a cached version exists.
-            min_heat_rate: lowest plausible heat rate, in mmBTU/MWh. Any MCOE
-                records with lower heat rates are presumed to be invalid, and
-                are discarded before returning.
-            min_cap_fact: minimum generator capacity factor. Generator records
-                with a lower capacity factor will be filtered out before
-                returning. This allows the user to exclude generators that
-                aren't being used enough to have valid.
-            min_fuel_cost_per_mwh: minimum fuel cost on a per MWh basis that is
-                required for a generator record to be considered valid. For
-                some reason there are now a large number of $0 fuel cost
-                records, which previously would have been NaN.
-            max_cap_fact: maximum generator capacity factor. Generator records
-                with a lower capacity factor will be filtered out before
-                returning. This allows the user to exclude generators that
-                aren't being used enough to have valid.
-            all_gens: Controls whether the output contains records for
-                all generators in the :ref:`generators_eia860` table, or only
-                those generators with associated MCOE data. True by default.
-            gens_cols: equal to the string "all", None, or a list of names of
-                column attributes to include from the :ref:`generators_eia860` table in
-                addition to the list of defined `DEFAULT_GENS_COLS` in the MCOE analysis
-                module. If "all", all columns from the generators table will be included.
-                By default, the `DEFAULT_GENS_COLS` defined in the MCOE analysis module
-                will be merged into the final MCOE output.
-
-        Returns:
-            A compilation of generator attributes, including fuel costs per MWh.
-        """
-        if update or self._dfs["mcoe"] is None:
-            self._dfs["mcoe"] = pudl.analysis.mcoe.mcoe(
-                self,
-                min_heat_rate=min_heat_rate,
-                min_fuel_cost_per_mwh=min_fuel_cost_per_mwh,
-                min_cap_fact=min_cap_fact,
-                max_cap_fact=max_cap_fact,
-                all_gens=all_gens,
-                gens_cols=gens_cols,
-            )
```

Comment:
I saw that at some point the filtering for valid heat rate, fuel price, and capacity values was being done in this explicitly defined MCOE PudlTabl function. I took out this function, but maybe it should happen here?